src/ssl.c

```diff
@@ -17813,16 +17831,56 @@ size_t wolfSSL_get_client_random(const WOLFSSL* ssl, unsigned char* out,
 #endif
     }
 
+    /* If a valid struct is provided with function pointers, will override
+       RAND_seed, bytes, cleanup, add, pseudo_bytes and status.  If a NULL
+       pointer is passed in, it will cancel any previous function overrides.
+
+       Returns WOLFSSL_SUCCESS on success, WOLFSSL_FAILURE on failure. */
+    int wolfSSL_RAND_set_rand_method(const WOLFSSL_RAND_METHOD *methods)
+    {
+        if (wc_LockMutex(&rand_methods_mutex) != 0) {
+            WOLFSSL_MSG("Lock rand methods mutex failed");
+            return WOLFSSL_FAILURE;
+        }
+        rand_methods = methods;
+        wc_UnLockMutex(&rand_methods_mutex);
+
+        return WOLFSSL_SUCCESS;
+    }
 
     int wolfSSL_RAND_status(void)
     {
+        int ret = WOLFSSL_FAILURE;
+
+        if (wc_LockMutex(&rand_methods_mutex) != 0) {
+            WOLFSSL_MSG("Lock rand methods mutex failed");
+            return WOLFSSL_FAILURE;
+        }
+        if (rand_methods != NULL && rand_methods->status != NULL) {
+            ret = rand_methods->status();
+            wc_UnLockMutex(&rand_methods_mutex);
+            return ret;
+        }
+        wc_UnLockMutex(&rand_methods_mutex);
+
         return WOLFSSL_SUCCESS;  /* wolfCrypt provides enough seed internally */
     }
 
 
     #ifndef NO_WOLFSSL_STUB
     void wolfSSL_RAND_add(const void* add, int len, double entropy)
     {
+        if (wc_LockMutex(&rand_methods_mutex) != 0) {
+            WOLFSSL_MSG("Lock rand methods mutex failed");
+            return;
+        }
+        if (rand_methods != NULL && rand_methods->add != NULL) {
+            rand_methods->add(add, len, entropy);
+            wc_UnLockMutex(&rand_methods_mutex);
+            return;
+        }
+        wc_UnLockMutex(&rand_methods_mutex);
+
         (void)add;
```

Comment:
I believe you can remove the unused variable code on `add`, `len`, `entropy`. Might be others for these functions.